LevelComponents/Level.cpp

```diff
@@ -327,71 +300,21 @@ namespace LevelComponents
     /// <summary>
     /// Distribute door data to every room.
     /// </summary>
-    void Level::RedistributeDoor()
+    void Level::SetLevelEntitySet()
     {
-        // Distribute door data to every room
-        for (unsigned int i = 0; i < doors.size(); ++i)
-        {
-            rooms[doors[i]->GetRoomID()]->AddDoor(doors[i]);
-        }
-
-        // Check if every Room have at least one Door, if not, set the entityset id to skip some problems
-        // the code is only for avoiding crash
+        // Check if every Room have at least one Door, if not, set a default entityset id to avoid some crashes
         for (unsigned int i = 0; i < rooms.size(); ++i)
         {
-            if (rooms[i]->CountDoors() == 0)
+            auto doorDataOfRoom = doorlist.GetDoorsByRoomID(rooms[i]->GetRoomID());
+            if (doorDataOfRoom.size())
             {
-                rooms[i]->SetCurrentEntitySet(37);
+                rooms[i]->SetCurrentEntitySet(doorDataOfRoom[0].EntitySetID);
             }
-        }
-    }
-
-    /// <summary>
-    /// Get the Doors data copy.
-    /// </summary>
-    /// <param name="roomId">
-    /// Room id of the Room the temp-Room copied from.
-    /// </param>
-    /// <remark>
-    /// only used to give data to the Room copy.
-    /// </remark>
-    std::vector<Door *> Level::GetRoomDoors(unsigned int roomId)
-    {
-        std::vector<Door *> roomDoors;
-        // Distribute door data
-        for (unsigned int i = 0; i < doors.size(); ++i)
-        {
-            if (doors[i]->GetRoomID() == (int) roomId)
+            else
             {
-                Door *newDoor = new Door(*doors[i]);
-                roomDoors.push_back(newDoor);
+                rooms[i]->SetCurrentEntitySet(37);
```

Comment:
What is 37?

Reply:
some default id for entityset, i moved it into the class as a init value for the variable